Add io.github.wdaniau.fstl

wdaniau opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [X] Please describe the application briefly. 
fstl-e is a fast stl file viewer. It is a fork of fstl (https://github.com/fstl-app/fstl) with some fancy enhancements.
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on io.github.wdaniau.fstl.yml:
```diff
@@ -0,0 +1,25 @@
+      - type: git
+        branch: fstl-e
+        url: https://github.com/wdaniau/fstl
+rename-desktop-file: fstlapp-fstl-e.desktop
+rename-icon: fstlapp-fstl-e
```
these two should be above `finish-args`

wdaniau replied:
ok, I'll do that.

hfiguiere commented on io.github.wdaniau.fstl.yml:
```diff
@@ -0,0 +1,25 @@
+    buildsystem: cmake
+    sources:
+      - type: git
+        branch: fstl-e
```
NEVER a git branch

wdaniau replied:
Well that's more a problem as fstl-e is my default branch on github but flatpak-builder use master ...

hfiguiere replied:
No, that's not a problem. A commit or a tag is REQUIRED.

The problem is setting a branch

wdaniau replied:
ok thanks

hfiguiere commented on io.github.wdaniau.fstl.yml:
```diff
@@ -0,0 +1,25 @@
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=xdg-documents
+  - --filesystem=home
```
`home` includes `xdg-documents`.

wdaniau replied:
noted, I'll suppress xdg-documents

hfiguiere replied:
it doesn't use portals? Since it's Qt based...

wdaniau replied:
I must admit I don't know ... ^^

wdaniau replied:
Ok after reading a bit. The only portals that are used in this app are the one described as transparently used on this page : https://docs.flatpak.org/en/latest/portals.html#portal-support-in-qt-and-kde

hfiguiere commented on io.github.wdaniau.fstl.yml:
```diff
@@ -0,0 +1,25 @@
+  - --filesystem=xdg-run/gvfs # make GnomeVFS accessible
+  - --filesystem=/run/media
+  - --filesystem=/media
+  - --device=all
```
why do you need `device=all`?

wdaniau replied:
you're right it's probably not necessary, I'll supress it.

hfiguiere commented:
bot, build io.github.wdaniau.fstl

wdaniau commented:
Hi, I added the missing developer tag in the metainfo file, changing the commit accordingly in manifest. I also added a flathub.json file to restrict the build to x86_64.

wdaniau commented:
And by the way I had forgotten to request a review after last commits, and was stupidly waiting for something to happen ^^

bbhtt commented:
bot, build io.github.wdaniau.fstl

wdaniau commented:
I tested the build and it is Ok. What should I do next?

wdaniau commented:
Is this ok? The label is still "awaiting-changes" even if the build was ok after I updated files.

bbhtt commented:
bot, build io.github.wdaniau.fstl

bbhtt commented:
/merge
